### PR TITLE
Start graql queries through graph

### PR DESCRIPTION
**Pull request by fppt**
- Create QueryBuilder interface
- Create graph.graql() method
- Start replacing use cases with graph.graql()

### CODE REVIEW

File: mindmaps-dist/src/test/java/io/mindmaps/dist/ExamplesTest.java [hunk 1]
```diff
@@ -35,7 +35,7 @@
 
 public class ExamplesTest {
 
-    private QueryBuilder qb;
+    private QueryBuilderImpl qb;
```
Comment (aelred) [inline mindmaps-dist/src/test/java/io/mindmaps/dist/ExamplesTest.java]: Maybe make these the interface where appropriate?

File: mindmaps-graql/src/main/java/io/mindmaps/graql/Graql.java [hunk 2]
```diff
@@ -43,16 +43,16 @@ private Graql() {}
...
     }
 
     /**
      * @param graph the graph to operate the query on
      * @return a query builder using the specified graph
      */
-    public static QueryBuilder withGraph(MindmapsGraph graph) {
-        return new QueryBuilder(graph);
+    public static QueryBuilderImpl withGraph(MindmapsGraph graph) {
+        return new QueryBuilderImpl(graph);
```
Comment (aelred) [inline mindmaps-graql/src/main/java/io/mindmaps/graql/Graql.java]: These should all return the interface, not the implementation

**Comment by fppt** [comment]
please retest this please

File: mindmaps-migration/base/src/main/java/io/mindmaps/migration/base/AbstractMigrator.java [hunk 3]
```diff
@@ -34,7 +34,7 @@
 public abstract class AbstractMigrator implements Migrator {
 
     public static final int BATCH_SIZE = 25;
-    public final QueryBuilder queryBuilder = Graql.withoutGraph();
+    public final QueryBuilderImpl queryBuilder = (QueryBuilderImpl) Graql.withoutGraph();
```
Comment (aelred) [inline mindmaps-migration/base/src/main/java/io/mindmaps/migration/base/AbstractMigrator.java]: Does this commit fix anything? It seemed fine before

Comment (fppt) [inline mindmaps-migration/base/src/main/java/io/mindmaps/migration/base/AbstractMigrator.java]: Another PR introduced this `registerMacro(Macro macro)` to `QueryBuilderImpl` and I did not add that to the interface. Hence the casting.

Comment (alexandraorth) [inline mindmaps-migration/base/src/main/java/io/mindmaps/migration/base/AbstractMigrator.java]: I'll make a pull req to add this method to the interface. There doesn't seem to be any reason not to.

**Pull request merged**